keyrings/alt/tests/test_pyfs.py

```diff
@@ -69,20 +98,35 @@ def test_handles_preexisting_keyring(self):
         pyf_keyring = pyfs.PlaintextKeyring(filename=self.keyring_filename)
         assert 'pwd1' == pyf_keyring.get_password('svc1', 'user1')
 
-    @pytest.fixture(autouse=True)
-    def remove_keyring_filename(self):
-        if os.path.exists(self.keyring_filename):
-            os.remove(self.keyring_filename)
+    @pytest.fixture(autouse=True, scope="class")
+    def _cleanup_for_fs(self):
+        yield
+        try:
+            shutil.rmtree(self.keyring_root)
+        except OSError:
+            e = sys.exc_info()[1]
+            if e.errno != errno.ENOENT:  # No such file or directory
+                raise
 
 
 class TestUnencryptedLocalPyfilesystemKeyringSubDir(PyFSBackend):
     """Test using local temp files with no encryption"""
 
-    keyring_dir = os.path.join(tempfile.mkdtemp(), 'more', 'sub', 'dirs')
+    keyring_root = tempfile.mkdtemp()
```

Comment:
This directory gets created unconditionally but is only cleaned up if the tests run. Better to put this creation behavior in the _cleanup_for_fs fixture before `yield` (and rename it to match new behavior).